pkg/runs/runsDownload.go

```diff
@@ -29,24 +30,103 @@ func DownloadArtifacts(
 
 	var err error = nil
 	var runs []galasaapi.Run
-	var artifactPaths []string
 
-	runs, err = GetRunsFromRestApi(runName, 0, 0, timeService, apiServerUrl)
+	if (err == nil) && (runName != "") {
+		err = ValidateRunName(runName)
+	}
+	if err == nil {
+		runs, err = GetRunsFromRestApi(runName, 0, 0, timeService, apiServerUrl)
+		if err == nil {
+			if len(runs) > 1 {
+				// get list of runs that are reRuns - get list of runs that are reRuns of each other
+				// create a map of lists of reRuns - key is queued time, value is the run
+				reRunsByQueuedTime := createMapOfReRuns(runs)
+
+				err = downloadReRunArtfifacts(reRunsByQueuedTime, forceDownload, fileSystem, apiServerUrl)
+
+			} else if len(runs) == 1 {
+				err = downloadArtifactsToDirectory(apiServerUrl, runName, runs[0], fileSystem, forceDownload)
+			} else {
+				err = console.WriteString("No artifacts to download for run: '" + runName + "'\n")
```

Comment:
This message should have a GAL number associated. It should be an information message.